WooCommerce/Classes/Bookings/BookingList/BookingListViewModel.swift

```diff
@@ -96,10 +104,14 @@ final class BookingListViewModel: ObservableObject {
     }
 
     /// Called when the user pulls down the list to refresh.
-    @MainActor
     func onRefreshAction() async {
+        await refreshCoordinator?.refreshAllLists()
+    }
+
+    @MainActor
+    func onRefreshSelfAction(reason: String? = nil) async {
         await withCheckedContinuation { continuation in
-            paginationTracker.resync(reason: Self.refreshCacheReason) {
+            paginationTracker.resync(reason: reason ?? Self.refreshCacheReason) {
```

Comment:
How about keeping the refresh cache as the default value for simplicity? And should we rename the method to `reloadData` for clarity?


```suggestion
    func reloadData(reason: String = Self.refreshCacheReason) async {
        await withCheckedContinuation { continuation in
            paginationTracker.resync(reason: reason) {
```